### DISCUSSION TITLE
Micropython equivalent of  __func__ or Python 2 im_func ?

### DISCUSSION THREAD

**Discussion by thisrandomuser** [micropython/micropython#10655] [General]
Does micropython have an equivalent of Python 3's __func__ or Python 2 im_func?

I'm trying to set up a call back in a class to another method in that class.

**Comment by dlech** [comment]
Can you explain more about what you are trying to do?

Given:

```python
class C:
    def m(self):
        ...
```

I assume `C.m` is what you are looking for?

**Reply by thisrandomuser** [reply]
Thank you for taking the time to reply 

I'm writing a class to hold my project specific code for dealing with mqtt messages. It sets up a new mqtt client by calling the functions in the umqttsimple.py library (https://mpython.readthedocs.io/en/master/library/mPython/umqtt.simple.html) which handles the generic mqtt functions. One of the steps is to set the function that the mqtt library calls back to when an mqtt message is received 
 (MQTTClient.set_callback(f)). When I have tried setting it using self.m (not C.m using your example), my function hasn't been getting called when subscribed mqtt messages are being published by other devices on my network. Should is be as simple as passing it Class.function instead of self.function? I will try it later when I can get to my computer.

**Reply by dlech** [reply]
Since `set_callback` expects a callback with the signature `f(topic, msg)`, `C.m` would only work if it has the signature `def m(self, msg)` where `C` is the topic object type. More likely you want something like:

```python
class C:
    def m(self, topic, msg):
        ...

c = C()

client.set_callback(c.m)
```

**Reply by thisrandomuser** [reply]
Thank you for the help. I still didn't manage to get it to work (probably user error), but I have an idea for a way to get around it when I get back to this project.
